enum.rb

```diff
@@ -0,0 +1,159 @@
+module Enumerable
+  def my_each
+    return unless block_given?
+
+    ar = self
+    ar.each do |el|
+      yield el
+    end
+  end
+
+  def my_each_with_index
```

Comment:
- [ ] should be identical and returns the same thing as ruby's [`each_with_index`](https://ruby-doc.org/core-2.6.4/Enumerable.html#method-i-each_with_index). ([Screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/advanced_building_blocks_enumerable/my_each_with_index.png) from [Odin](https://www.theodinproject.com/courses/ruby-programming/lessons/advanced-building-blocks#assignment-2))
- [ ] calls the given block once for each element in self
```ruby 
my_each_output = ''
block = proc { |num, idx| my_each_output += "Num: #{num}, idx: #{idx}\n" }
array.each_with_index(&block)
each_output = my_each_output.dup
my_each_output = ''
array.my_each_with_index(&block)
my_each_output == each_output #true
```
- [ ] returns an enumerator if no block is given
```ruby
array.my_each_with_index == Enumerator #true
```

Reply:
### Above edge cases are still failing.